portability-core/src/main/java/org/dataportabilityproject/serviceProviders/instagram/InstagramServiceProvider.java

```diff
@@ -73,10 +87,9 @@ public OnlineAuthDataGenerator getOnlineAuthDataGenerator(PortableDataType dataT
       PortableDataType type,
       AuthData authData,
       JobDataCache jobDataCache) throws IOException {
-    if (type == PortableDataType.PHOTOS) {
-      return new InstagramPhotoService(((InstagramOauthData) authData));
-    }
-    throw new IllegalStateException("Instagram doesn't support exporting: " + type);
+    Preconditions
+        .checkArgument(exportTypes.contains(type), "Instagram doesn't support exporting %s", type);
```

Comment:
Let's surround %s in brackets or put it after a semicolon to highlight it is dynamically populated

export [%s] type or export type: %s"

Reply:
Done